components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/KeyStoreAdmin.java

```diff
@@ -785,6 +813,64 @@ public PaginatedKeyStoreData getPaginatedKeystoreInfo(String keyStoreName, int p
 
     }
 
+    /**
+     * This method will add 1. Certificate aliases 2. Private key alise 3. Private key value to a
+     * given keystore with certificate and keystore filtering.
+     *
+     * @param keyStoreName The name of the keystore.
+     * @param pageNumber   Page number.
+     * @param filterString Filter text.
+     * @return Instance of KeyStoreData.
+     * @throws SecurityConfigException will be thrown.
+     */
+    public PaginatedKeyStoreData getFilteredPaginatedKeystoreInfo(String keyStoreName, int pageNumber,
+                                                                  String filterString) throws SecurityConfigException {
+
+        if (keyStoreName == null) {
+            throw new SecurityConfigException("Keystore name cannot be null.");
+        }
+
+        try {
```

Comment:
Simplify the two methods getPaginatedKeystoreInfo and getFilteredPaginatedKeystoreInfo and remove the redundant logic.

Reply:
Resolved with PR https://github.com/wso2/carbon-identity-framework/pull/4125